### PR TITLE
[SERV-570] Add test Solr index

### PR DESCRIPTION
**Pull request by markmatney**
To test, open two terminals:

1. In the first, run:

    ```bash
    mvn clean package docker:build docker:run -Dmaven.test.skip -Ddocker.showLogs
    ```

2. In the second, run:

    ```bash
    curl -X POST 'http://localhost:${test.solr.port}/solr/prl/update?wt=json&commitWithin=1000&overwrite=true' \
        --data '[ { "message": "Hello, World!" } ]'
    curl 'http://localhost:${test.solr.port}/solr/prl/select?q=*:*'
    ```

Tested with Maven 3.8.5.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,6 +51,7 @@
     <!-- Docker component versions -->
     <docker.alpine.version>3.16.2</docker.alpine.version>
     <postgres.version>12.12-alpine</postgres.version>
+    <solr.version>8</solr.version>
```
Comment (ksclarke) [inline pom.xml:54]: We're using an older version because we determined there were API changes that would break the front-end if we upgraded?

**Review by ksclarke — DISMISSED** [review]

File: pom.xml (hunk 1 above)
Comment (markmatney) [reply pom.xml:54]: I figured we'd use 8 for consistency, since we just migrated several apps to 8

Comment (ksclarke) [reply pom.xml:54]: Up to you. Solr describes 8 as "Previous major version _may sometimes_ [emphasis mine] receive critical bugfix releases"

Comment (markmatney) [reply pom.xml:54]: The Postgres version specified in the POM is older too, btw, which I assumed we were also doing for consistency with other apps.

Comment (markmatney) [reply pom.xml:54]: @cachemeoutside do you see any other reason (deployment-related or otherwise) why we shouldn't use Solr 9?

Comment (ksclarke) [reply pom.xml:54]: Is this project going to use a centralized postgres or spin up its own? For some reason I thought we were spinning up our own Solr but using the centralized postgres(?)

Comment (cachemeoutside) [reply pom.xml:54]: I have no objection for solr 9. I don't think we plan on running a centralized solr going forward due to prior issues with updating

Comment (cachemeoutside) [reply pom.xml:54]: Solr 9 appears to be the latest stable version. I'd say switch to it if you can.

Comment (markmatney) [reply pom.xml]: Switched

**Review by ksclarke — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Review by cachemeoutside — APPROVED** [review]

**Pull request merged**